wordpress-mobile/WordPress-Android
# [Compile Warnings As Errors] WordPress Module - Resolve `PreferenceManager` Deprecated Warnings

Parent: #17173
Partially Closes: #17181

This PR resolves all `PreferenceManager` related `Deprecated` warnings for the `WordPress` module and beyond (`Analytics` and `Editor` modules):

Modules:
- Analytics: [Resolve preference manager deprecated warnings for analytics.](https://github.com/wordpress-mobile/WordPress-Android/commit/2fae16df781a65c2a8017a2ddfb5d4c77c3fd963)
- Editor: [Resolve preference manager deprecated warnings for editor.](https://github.com/wordpress-mobile/WordPress-Android/commit/b5d47b56070d11f50ca796787737b9dd9dabf3ee)
- WordPress: [Resolve preference manager deprecated warnings for wordpress.](https://github.com/wordpress-mobile/WordPress-Android/commit/5f354a202dbe4bf1fd37ea8ad9e0b33e0d421469)

-----

PS: ~~@RenanLukas~~ @AjeshRPai I added you as the main reviewer, that is, in addition to @wordpress-mobile/apps-infrastructure team itself, but randomly, since I want someone from the `WordPress Android` team to primarily sign-off on that change. 🥇

FYI: I am going to randomly add more of you in those PRs that will follow, just so you become more aware of this change and how close we are on enabling `allWarningsAsErrors` by default everywhere. 🎉

-----

To test:

- Smoke test both, the `WordPress` and `Jetpack` apps, and see if their settings and preference related functionalities are working as expected. For example, and per module, you could test the following:
  - Analytics: Switch on and off the `Collect Information` privacy setting and verify that the `Logcat` is showing/hiding the `🔵 Tracked: xyz` related information. To find this setting go to: `My Site` tab -> `Me` options -> `App Settings` -> `Privacy settings`
  - Editor: Switch on and off the `Use Block Editor` site setting and verify that the `Dialog` that appears when using the legacy editor is showing/hiding. The dialog's title should be `Try the new Block Editor`. To find this setting go to: `My Site` tab -> `Site Settings` configuration
  - WordPress: Update the interface language and verify that the app's language changes accordingly. To find this setting got to: `My Site` tab -> `Me` options -> `App Settings` -> `Interface Language`

PS: The above testing should be enough to verify that all other setting related affecting changes are also working as expected.

-----

## Regression Notes
1. Potential unintended areas of impact

The settings and preference related functionalities are not working as expected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
